### ISSUE TITLE
[Feature Request] Improve a-la-carte experience

### ISSUE THREAD

**Issue by yordis** [vuetifyjs/vuetify#4234]
### Problem to solve
A-la-carte users right now can't benefit from tooling like IDEs and Editors that are smart importing dependencies.

Because right now everything is exported throw the `default` scope of the module the IDEs can't figure out which code you are referring to.

This could be quite hard to memorize where is each component but at least by inferring the name from the component example, for example `v-btn` would make sense to be `VBtn`, we could speed up the productivity and no bother by those imports.

In the long term will help users to ship a well performance application by letting the tools figuring out the tree-shaking.

### Proposed solution
By exporting the components using a name like

```typescript
// ../VContainer/VContainer.ts

const VContainer = { };

export VContainer;
```

At least you get some autocomplete from IDEs when you use `VContainer` after the indexing.

The downside is that most of the time (I am guessing) people would expect a Vue component to be export in the default which could be fixable if each component has it own folder and you re-export the component to the default, like

```typescript
// ../VContainer/index.ts
export { VContainer as default } from "./VContainer";
```
Doing that approach you have default exports and named exports but we should always have the named exports at least.

**Comment by jacekkarczmarczyk** [comment]
Sounds for me like duplicate of https://github.com/vuetifyjs/vuetify/issues/4150

**Comment by yordis** [comment]
@jacekkarczmarczyk yeah it is like an add-on I would say, my main concern is that `default` exporting makes it hard to use a-la-carte because IDEs can't figure out which component you are referring to.

Using named export would fix it a little bit.

**Comment by mrts** [comment]
Looks like a la carte components might not be needed in the longer perspective, see #4845

**Comment by KaelWD** [comment]
Resolved in 1.3 with c3c61a9fab5dc286b476cbf1e244cee190ac8b1f
